src/main/java/org/prebid/server/cache/CoreCacheService.java

```diff
@@ -385,26 +398,34 @@ private CachedCreative createJsonPutObjectOpenrtb(CacheBid cacheBid,
             bidObjectNode.put(BID_WURL_ATTRIBUTE, eventUrl);
         }
 
+        final String resolvedCacheKey = resolveCacheKey(accountId);
+
         final BidPutObject payload = BidPutObject.builder()
                 .aid(eventsContext.getAuctionId())
                 .type("json")
+                .key(resolvedCacheKey)
                 .value(bidObjectNode)
                 .ttlseconds(cacheBid.getTtl())
                 .build();
 
         return CachedCreative.of(payload, creativeSizeFromAdm(bid.getAdm()));
     }
 
-    private CachedCreative createXmlPutObjectOpenrtb(CacheBid cacheBid, String requestId, String hbCacheId) {
+    private CachedCreative createXmlPutObjectOpenrtb(CacheBid cacheBid,
+                                                     String requestId,
+                                                     String hbCacheId,
+                                                     String accountId) {
```

Comment:
Nitpick. Please, add empty line after method signature. We always do it for separation of signature and method body.